UI/src/components/widgets/codeanalysis/testdetails.js

```diff
@@ -0,0 +1,72 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('devops-dashboard')
+        .controller('TestDetailsController', TestDetailsController);
+
+    TestDetailsController.$inject = ['$scope','$modalInstance', 'testResult', 'DASH_STATUS'];
+    function TestDetailsController($scope, $modalInstance, testResult, DASH_STATUS) {
+        /*jshint validthis:true */
+        var ctrl = this;
+
+        ctrl.statuses = DASH_STATUS;
+        ctrl.testResult = testResult;
+        ctrl.duration = msToTime(testResult.duration);
+        ctrl.close = close;
+
+        function close() {
+            $modalInstance.dismiss('close');
+        }
+
+        $scope.showCapabilityDetail = function (capability) {
+            if ($scope.activeCapability != capability.description) {
+                $scope.activeCapability = capability.description;
+            }
+            else {
+                $scope.activeCapability = null;
+            }
+        };
+        $scope.showTestSuiteDetail = function (testSuite) {
+            if ($scope.activeSuite != testSuite.description) {
+                $scope.activeSuite = testSuite.description;
+            }
+            else {
+                $scope.activeSuite = null;
+            }
+        };
+        $scope.showTestCaseDetail = function (testCase) {
+            if ($scope.activeCase != testCase.description) {
+                $scope.activeCase = testCase.description;
+            }
+            else {
+                $scope.activeCase = null;
+            }
+        };
+
+        $scope.showStatusIcon =
+        function showStatusIcon(item) {
+            if (item.status.toLowerCase() == 'success') {
+                return 'ok';
+            } else if (item.status.toLowerCase() == 'skipped') {
+                return 'warning';
+            } else {
+                return 'error';
+            }
+        };
+
+        function msToTime(duration) {
```

Comment:
Can be. Added an enhancement issue. Little more involved than what I thought it would take.